src/components/views/DownloadModModal.vue

```diff
@@ -92,28 +88,36 @@ let assignId = 0;
         public static async downloadSpecific(
             profile: Profile,
             combo: ThunderstoreCombo,
-            ignoreCache: boolean
+            ignoreCache: boolean,
+            store: Store<any>
         ): Promise<void> {
             return new Promise((resolve, reject) => {
                 const tsMod = combo.getMod();
                 const tsVersion = combo.getVersion();
-                const currentAssignId = assignId++;
+
+                store.commit('download/increaseAssignId');
+                const currentAssignId = store.state.download.assignId;
+
                 const progressObject = {
                     progress: 0,
                     initialMods: [`${tsMod.getName()} (${tsVersion.getVersionNumber().toString()})`],
                     modName: '',
                     assignId: currentAssignId,
                     failed: false,
                 };
-                DownloadModModal.allVersions.push([currentAssignId, progressObject]);
+
+                store.commit('download/pushDownloadObjectToAllVersions', progressObject);
                 setTimeout(() => {
                     ThunderstoreDownloaderProvider.instance.download(profile.asImmutableProfile(), tsMod, tsVersion, ignoreCache, (progress: number, modName: string, status: number, err: R2Error | null) => {
-                        const assignIndex = DownloadModModal.allVersions.findIndex(([number, val]) => number === currentAssignId);
+                        const assignIndex = store.state.download.allVersions.findIndex(([number, val]: [number, DownloadProgress]) => number === currentAssignId);
                         if (status === StatusEnum.FAILURE) {
                             if (err !== null) {
-                                const existing = DownloadModModal.allVersions[assignIndex]
+                                const existing = store.state.download.allVersions[assignIndex]
                                 existing[1].failed = true;
-                                DownloadModModal.allVersions[assignIndex] = [currentAssignId, existing[1]];
+                                store.commit('download/updateDownloadObject', {
+                                    assignId: assignIndex,
+                                    downloadObject: [currentAssignId, existing[1]]
```

Comment:
The argument type here is not what you've defined the downloadObject to be in the Vuex store (it expects only DownloadProgress object. This can be fixed either on the call sites or on the Vuex method. Do note that there are multiple call sites to fix if you choose to go that route.

I know the original code is confusing but be careful about the arguments of Vuex methods as the type support doesn't really exist.